Frontend/Pages/Projects/LegalRequirements/DiocesanConsent.cshtml.cs

```diff
@@ -0,0 +1,96 @@
+using Data;
+using Frontend.Models.Benefits;
+using Frontend.Models.Forms;
+using Frontend.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Frontend.ExtensionMethods;
+using Frontend.Models.LegalRequirements;
+
+namespace Frontend.Pages.Projects.LegalRequirements
+{
+    public class DiocesanConsentModel : CommonPageModel
+    {
+        private readonly IProjects _projects;
+
+        public DiocesanConsentModel(IProjects projects)
+        {
+            _projects = projects;
+        }
+
+        [BindProperty] public DiocesanConsentViewModel DiocesanConsentViewModel { get; set; } = new DiocesanConsentViewModel();
+
+        public IList<RadioButtonViewModel> RadioButtonsYesNoNotApplicable { get; set; }
+
+
+        public async Task<IActionResult> OnGetAsync()
+        {
+            var project = await _projects.GetByUrn(Urn);
+            IncomingTrustName = project.Result.IncomingTrustName;
+
+            RadioButtonsYesNoNotApplicable = GetRadioButtons(project.Result.LegalRequirements.DiocesanConsent.ToDescription());
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            var project = await _projects.GetByUrn(Urn);
+
+            if (ModelState.IsValid is false)
+            {
+                RadioButtonsYesNoNotApplicable = GetRadioButtons(project.Result.LegalRequirements.DiocesanConsent.ToDescription());
+                return Page();
+            }
+
```

Comment:
This file has a few double line breaks as well.

Reply:
Done